__tests__/app.test.js

```diff
@@ -3,28 +3,72 @@ const app = require("../app");
 
 beforeEach(() => {
     return seed(testData);
 });
 
 afterAll(() => {
     return db.end();
-    // return seed(testData).then(() => db.end());
 });
 
-describe("GET /api/topics", () => {
-    test("should return 200 and array of objects which they should have slug and description properties", () => {
-        return request(app)
-            .get("/api/topics")
-            .expect(200)
-            .then(({ body }) => {
-                const topics = body.topics;
-                expect(topics).toHaveLength(3);
-                topics.forEach((topic) => {
-                    expect(topic).toHaveProperty("slug", expect.any(String));
-                    expect(topic).toHaveProperty("description", expect.any(String));
+describe("GET / POST / PUT / PATCH / DELETE ", () => {
+    describe("GET", () => {
+        test("should return 200 and array of objects which they should have slug and description properties", () => {
+            return request(app)
+                .get("/api/topics")
+                .expect(200)
+                .then(({ body }) => {
+                    const topics = body.topics;
+                    expect(topics).toHaveLength(3);
+                    topics.forEach((topic) => {
+                        expect(topic).toHaveProperty("slug", expect.any(String));
+                        expect(topic).toHaveProperty("description", expect.any(String));
+                    });
                 });
-            });
+        });
+        test("should return 404", () => {
+            return request(app)
+                .get("/api/wrongTopic")
+                .expect(404)
+                .then(({ body }) => {
+                    const topics = body;
+                    expect(topics.msg).toBe("Not found!");
+                });
+        });
+        test("/api should return 200: and correct response json obj", () => {
+            const listOfEndpoints = Object.keys(descriptionTemplateData);
+            return request
+                .agent(app)
+                .get("/api")
+                .expect(200)
+                .then(({ body }) => {
+                    const descriptions = body.descriptions;
+
+                    listOfEndpoints.forEach((endpoint) => {
+                        if (descriptions.hasOwnProperty(endpoint)) {
+                            expect(descriptions[endpoint]).toHaveProperty(
+                                "description",
+                                expect.any(String)
+                            );
+                            expect(descriptions[endpoint]).toHaveProperty(
+                                "queries",
+                                expect.any(Array)
+                            );
+                            expect(descriptions[endpoint]).toHaveProperty(
+                                "exampleResponse",
+                                expect.any(Object)
+                            );
+
+                            if (endpoint === "GET /api/topics") {
```

Comment:
This test is overcomplicating things a bit. For ticket 3, we can simply require in the endpoints JSON file and then expect it to be sent as the response for that endpoint. 👍